modules/drivers/redshift/test/metabase/driver/redshift_test.clj

```diff
@@ -192,3 +196,29 @@
                    (map
                     (partial into {})
                     (db/select [Field :name :database_type :base_type] :table_id table-id))))))))))
+
+(defn- assert-jdbc-url-fetch-size [db fetch-size]
+  (let [ds   (sql-jdbc.execute/datasource db)
+        conn (.getConnection ds)
+        md   (.getMetaData conn)
+        url  (.getURL md)]
+    (is (str/includes? url (str "defaultRowFetchSize=" fetch-size)))))
+
+(deftest test-jdbc-fetch-size
+  (testing "Redshift JDBC fetch size is set correctly in PreparedStatement"
+    (mt/test-driver :redshift
+      (assert-jdbc-url-fetch-size (mt/db) 5000) ; the default value should always be picked up if nothing is set
+      (with-redefs [env/env (assoc env/env :mb-redshift-fetch-size "14")]
+        ; we have to create a new DB in order to pick up the change to the environment here
+        (mt/with-temp Database [db {:engine :redshift, :details (:details (mt/db))}]
+          (mt/with-db db
+            ;; make sure the JDBC URL has the defaultRowFetchSize parameter set correctly
+            (assert-jdbc-url-fetch-size db 14)
+            ;; now, actually run a query and see if the PreparedStatement has the right fetchSize set
+            (qp.store/with-store
+              (qp.store/fetch-and-store-database! (u/the-id db))
+              (let [max-rows 10000
+                    q        "SELECT 1"
+                    ctx      (context.default/default-context)
+                    rs-fn    (fn [rs] (is (= 14 (.getFetchSize (.getStatement rs)))))]
+                   (#'sql-jdbc.execute/run-fn-with-result-set :redshift q {} max-rows ctx rs-fn)))))))))
```

Comment:
Instead of defining a new function here just for this one test situation, why not just with-redefs `sql-jdbc.execute/execute-query!` or `execute-reducible-query`? See for example `metabase.query-processor.middleware.cache-test/export-test`. I'd rather not make code more complicated in a general namespace like `sql-jdbc.execute` for the sake of one test in one driver somewhere. It's not a scalable way to do things

Reply:
OK, reworked the test to redef `reducible-rows`, which allows us to capture what we need there.